### DISCUSSION TITLE
How will peeky handle browser testing?

### DISCUSSION THREAD

**Discussion by edimitchel** [Akryum/peeky#12] [Ideas]
Component testing is a most needed feature for peeky, but would tests be rendered in a virtual DOM or a real one ?

Using jsdom for a virtual rendering? **Quick and interoperable** but also **not as real as a browser** based rendering.
Using web driver? **Real reproduction and using real browser** but heavy to configure and need infrastructure.
Using browser embedded? **JS driven and using installed browser** but slower (?).
_Special option_: Embedded Web View from Tauri (which would be limited to the render engine of the machine host

Could discuss together to find a nice solution since this project is at his start-up!

**Comment by Akryum** [comment]
I think a good first step would be using jsdom since one of the promise of peeky is being fast. BTW is there something faster than jsdom?

**Reply by edimitchel** [reply]
Don't know something faster, but may be, make the usage of JSDOM through a facade for letting peeky to operate things with another rendering solution in future ?

**Reply by Akryum** [reply]
Maybe this is faster? https://github.com/WebReflection/linkedom#readme

**Reply by edimitchel** [reply]
Interesting !

**Reply by edimitchel** [reply]
Happy DOM
https://github.com/capricorn86/happy-dom

**Comment by N0tExisting** [comment]
Because this kinda fits into the toppic:
### Does Peeky use a package's browser exports?
